relay-server/src/actors/processor.rs

```diff
@@ -2064,83 +2057,74 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
-    /// Extract metrics for transaction events with breakdowns and measurements.
-    fn extract_transaction_metrics(
-        &self,
-        state: &mut ProcessEnvelopeState,
-    ) -> Result<(), ProcessingError> {
-        if state.transaction_metrics_extracted {
-            // Nothing to do here.
-            return Ok(());
-        }
+    /// Extract metrics from all envelope items.
+    ///
+    /// Caveats:
+    ///  - This functionality is incomplete. At this point, extraction is implemented only for
+    ///    transaction events.
+    fn extract_metrics(&self, state: &mut ProcessEnvelopeState) -> Result<(), ProcessingError> {
+        // TODO: Make span metrics extraction immutable
+        if let Some(event) = state.event.value_mut() {
+            if state.transaction_metrics_extracted {
```

Comment:
Should we go ahead and rename this field to `event_metrics_extracted`?

Reply:
Yeah, let's go ahead with this.